packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1723,9 +1799,9 @@ export class LinuxBuildImage implements IBuildImage {
   /** The Amazon Linux 2 x86_64 standard image, version `3.0`. */
   public static readonly AMAZON_LINUX_2_3 = LinuxBuildImage.codeBuildImage('aws/codebuild/amazonlinux2-x86_64-standard:3.0');
 
-  public static readonly AMAZON_LINUX_2_ARM: IBuildImage = new ArmBuildImage('aws/codebuild/amazonlinux2-aarch64-standard:1.0');
+  public static readonly AMAZON_LINUX_2_ARM = LinuxArmBuildImage.fromCodeBuildImageId('aws/codebuild/amazonlinux2-aarch64-standard:1.0');
   /** Image "aws/codebuild/amazonlinux2-aarch64-standard:2.0". */
-  public static readonly AMAZON_LINUX_2_ARM_2: IBuildImage = new ArmBuildImage('aws/codebuild/amazonlinux2-aarch64-standard:2.0');
+  public static readonly AMAZON_LINUX_2_ARM_2 = LinuxArmBuildImage.fromCodeBuildImageId('aws/codebuild/amazonlinux2-aarch64-standard:2.0');
```

Comment:
Feel free to deprecate these 2, pointing to the new constants in the new class.

Reply:
Slight hiccup with these and moving the LinuxArmBuildImage to its own file: The LinuxArmBuildImage needs to include things from the project.ts, but if moved out, these would need to include the new linux-arm-build-image.ts. Does that create a circular dependency?

If so, is there any other way to solve this than throwing and making this a breaking change?

Reply:
It could, possibly. Can you try the move, and see if you get an error?

The way to get rid of these cycles is to move the import to the end of the file, instead of the beginning. [Like this](https://github.com/aws/aws-cdk/blob/3382e997067f7e9f98ba98f618402b48bea838b9/packages/%40aws-cdk/core/lib/cfn-element.ts#L165-L167).

Reply:
I'm having trouble with the circular dependency and the unit tests. Specifically, unrelated unit tests, such as build-spec.test.js, fail. Guessing because of all of them importing * from ../lib.

Importing LinuxArmBuildImage in the bottom of project.ts results in:
```
ReferenceError: Cannot access 'linux_arm_build_image_1' before initialization
@aws-cdk/aws-codebuild:     > 1701 |   public static readonly AMAZON_LINUX_2_ARM = LinuxArmBuildImage.AMAZON_LINUX_2_STANDARD_1_0;
```
while importing the LinuxArmBuildImage up top of project.ts results in:
```ReferenceError: Cannot access 'project_1' before initialization
@aws-cdk/aws-codebuild:       65 |     return new LinuxArmBuildImage({
@aws-cdk/aws-codebuild:       66 |       imageId: id,
@aws-cdk/aws-codebuild:     > 67 |       imagePullPrincipalType: ImagePullPrincipalType.CODEBUILD
```
Any ideas how to fix the import flow so the tests that do the import * from ../lib would pass?

Reply:
The build is now successful. But I need your input on if the way it is fixed in project.ts is feasible.

That is, since the project.ts exports the LinuxArmBuildImage depends on are declared further up, and the only reason to include the LinuxArmBuildImage in the project.ts is the ```@depracted``` images, having the import just before their use (on line 1672) fixes the circular dependency.This is due to when the project.ts includes the LinuxArmBuildImage, all the parts of project.ts the LinuxArmBuildImage uses have been created so all is kosher.

My thinking is this would be okay as I would think down the line the deprecated constants would be removed, along with that import. If this does not seem proper, let me know how else to split the file out and have it build properly.

Reply:
> The build is now successful. But I need your input on if the way it is fixed in project.ts is feasible.
> 
> That is, since the project.ts exports the LinuxArmBuildImage depends on are declared further up, and the only reason to include the LinuxArmBuildImage in the project.ts is the `@depracted` images, having the import just before their use (on line 1672) fixes the circular dependency.This is due to when the project.ts includes the LinuxArmBuildImage, all the parts of project.ts the LinuxArmBuildImage uses have been created so all is kosher.
> 
> My thinking is this would be okay as I would think down the line the deprecated constants would be removed, along with that import. If this does not seem proper, let me know how else to split the file out and have it build properly.

I'm fine with this, as long as the tests work with the qualified import - see https://github.com/aws/aws-cdk/pull/19052#discussion_r820003014.